core/src/main/java/org/infinispan/util/concurrent/locks/OwnableReentrantLock.java

```diff
@@ -40,6 +42,7 @@ protected final Object currentRequestor() {
    }
 
    private void setCurrentRequestor(Object requestor) {
+      assert requestorOnStack.get() == null;
```

Comment:
During tests, assertions are enabled. Of course, in production this would not be called (and that's the idea!)